generic_scripts/nlk.py

```diff
@@ -0,0 +1,304 @@
+            wire_positions[:, 1::2] = _np.array([s1*10, s2*5.85])
+            wire_positions *= 1e-3
+
+        # Creating wires
+        currents = _np.zeros(8)
+        currents[::2] = -current
+        currents[1::2] = current
+
+        self._wires = [
+            Wire() for i in range(8)]
+        self.positions = wire_positions
+        self.currents = currents
+
+    @property
+    def wires(self):
+        """List with the Wires objects that compose the NLK.
+        Returns
+        -------
+        list of Wires objects.
+        """
+        return self._wires
+
+    @property
+    def positions(self):
+        """Position coordinates of the wires.
+
+        Returns
+        -------
+        numpy.array
+            Returns a numpy.array of shape (2, 8) with the wires coordinates.
+        """
+        return _np.array([wi.position for wi in self.wires]).T
+
+    @positions.setter
+    def positions(self, wires_positions):
+        """."""
+        for i in range(wires_positions.shape[1]):
+            self.wires[i].position = wires_positions[:, i]
+
+    @property
+    def currents(self):
+        """Currents of the NLK wires in [A].
+
+        Returns
+        -------
+        numpy.array
+            Single dimension array with the currents of each wire in NLK.
+        """
+        return _np.array([wi.current for wi in self._wires])
+
+    @currents.setter
+    def currents(self, wires_currents):
+        for i, current in enumerate(wires_currents):
+            self._wires[i].current = current
+
+    def calc_magnetic_field(self, r):
+        """Computes the magnetic field generated by NLK in a set of points.
+
+        Parameters
+        ----------
+        r : numpy.array
+            Array of shape (2 x N) with the coordinates [[x1, x2, ... , xN],
+            [y1, y2, ... , yN]] where the field will be returned.
+
+        Returns
+        -------
+        numpy.array
+            Array with the same shape of r, containing the magnetic field
+            components in [T] at each point in r.
+
+        Raises
+        ------
+        ValueError
+            If the input r hasn't the shape of type (2 X N).
+        """
+        if r.shape[0] != 2:
+            raise ValueError(
+                "The input position vector r must have shape of type (2, N)")
+        mag_field = _np.zeros(r.shape)
+        for wire in self.wires:
+            mag_field += wire.calc_magnetic_field(r)
+        return mag_field
+
+    def get_vertical_magnetic_field(self, y0=0):
+        """NLK vertical field at a horizontal plane y=y0 for x ∈ [-12, 12] mm.
+
+        Parameters
+        ----------
+        y0 : float, optional
+            y coordinate of the horizontal plane.
+
+        Returns
+        -------
+        x_pos: numpy.array
+            Values of x where the field was computed.
+
+        fieldy: numpy.array
+            Vertical field in [T] at the plane y=y0.
+        """
+        x_pos = _np.linspace(-12, 12)[None, :]*1e-3
```

Comment:
I think you can simplify this to:
```
x_pos = _np.linspace(-12, 12)*1e-3
y_pos = _np.ones(x_pos.shape) * y0
r = _np.vstack([x_pos, y_pos])
fieldy = self.calc_magnetic_field(r)[1]
return x_pos, fieldy
```

Reply:
Yes, in this way became more simple. These lines are a residue of when I was mistakenly trying to use column vectors everywhere.